cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/meta/AbstractRemoteStoreHandler.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.gateway.handlers.meta;
+
+import co.cask.cdap.common.BadRequestException;
+import co.cask.cdap.internal.app.store.remote.MethodArgument;
+import co.cask.http.AbstractHttpHandler;
+import com.google.common.base.Charsets;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.reflect.TypeToken;
+import org.jboss.netty.handler.codec.http.HttpRequest;
+
+import java.lang.reflect.Type;
+import java.util.Iterator;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * Implements common functionality for reading method arguments and deserializing them.
+ */
+public class AbstractRemoteStoreHandler extends AbstractHttpHandler {
```

Comment:
Since this is not specific to "Store", should make this more generic. ``AbstractRemoteSystemOpsHandler``? ``AbstractRemoteProgramOpsHandler``